addons/website_sale_stock_wishlist/static/src/js/website_sale.js

```diff
@@ -0,0 +1,67 @@
+/** @odoo-module **/
+
+import WebsiteSale from 'website_sale_stock.website_sale';
+import {is_email} from 'web.utils';
+
+WebsiteSale.include({
+    events: Object.assign({}, WebsiteSale.prototype.events, {
+        'click #add_stock_email_notification_wishlist_button': '_onClickAddStockEmailNotificationWishlistMessage',
+        'click #add_stock_email_notification_wishlist_form_button': '_onSubmitAddStockEmailNotificationWishlistForm',
+    }),
+
+    _onClickAddStockEmailNotificationWishlistMessage: function (ev) {
+        //hide the message, display the input box
+        ev.currentTarget.classList.add('d-none');
+        const form = ev.currentTarget.parentElement.querySelector('#add_stock_email_notification_wishlist_form');
+
+        form.classList.remove('d-none');
+
+    },
+
+    _onSubmitAddStockEmailNotificationWishlistForm: function (ev) {
+        const form = ev.currentTarget.closest('#add_stock_email_notification_wishlist_form');
+        const product_id = JSON.parse(ev.currentTarget.closest('tr').getAttribute('data-product-tracking-info')).item_id;
+        const email = form.querySelector('input[name="email"]').value;
+        const incorrect_icon = form.querySelector('#add_stock_email_notification_product_input_incorrect');
+        if (!is_email(email)) {
+            incorrect_icon.classList.remove('d-none');
+            return
+        }
+        this._rpc({
+            route: "/shop/add_stock_email_notification",
+            params: {
+                product_id,
+                email
+            },
+        }).then(function (data) {
+            if (data === '400 Bad Request: Invalid Email'){
```

Comment:
I am not a fan of this condition.
I think we never use `data`. Could we check if `data` is a string instead and consider that if `data` is a string, it's an error?

Reply:
<img width="878" alt="Screenshot 2022-05-31 at 12 54 51" src="https://user-images.githubusercontent.com/101560055/171157869-88850a30-0f54-4ac8-8836-dd4fd99e1862.png">

The thing is that here, there is no error, the server returns a string which contains `"product.wishlist(52,)"`. Is there a way to check the status code ?

Reply:
Quite honestly it could be simpler, do we actually need the new record in this case? I don't think so..
Either way we can return something as simple as `{'error': message}` in case of error and `{'wish': wish}` in case of success.
The status code does not matter.